zb-db/src/main/java/io/zeebe/db/impl/rocksdb/ZeebeRocksDbFactory.java

```diff
@@ -181,7 +189,7 @@ private ColumnFamilyOptions createDefaultColumnFamilyOptions(
         // merge at least 3 memtables per L0 file, otherwise all memtables are flushed as individual
         // files
         // this is also a candidate for tuning, it was a rough guess
-        .setMinWriteBufferNumberToMerge(Math.min(3, maxConcurrentMemtableCount))
+        .setMinWriteBufferNumberToMerge(rocksDbConfiguration.getMinWriteBufferNumberToMerge())
```

Comment:
So it is no longer related to the maxConcurrentMemtableCount is this good?

Reply:
What I found in general is that it should be about half of the max concurrent memtable, but I didn't investigate if setting higher would improve things. Setting it lower (e.g. 4 memtables, 1 min to merge) decreased throughput noticeably. I still don't know what the correct relation is, so by default it's already half, and then if someone changes the default settings they have to be careful and measure their results.